### PR TITLE
Implement sitemap generation using the 'sitemap_generator' gem

### PR DESCRIPTION
**Pull request by joancodes**
# Notable Changes

* This PR aims to add sitemap generation for hotwire.io to improve SEO by providing search engines with structured access to the site's pages.

Issue #47 

## Screenshots (Optional)

* N/A

## Todo

* N/A

### CODE REVIEW

**Comment by joancodes** [comment]
Hi @marcoroth, this is ready for review.

File: config/sitemap.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Set the host name for URL creation
+SitemapGenerator::Sitemap.default_host = "https://www.hotwire.io"
```
Comment (marcoroth) [inline config/sitemap.rb:2]: Can we use the non-www version here:

```suggestion
SitemapGenerator::Sitemap.default_host = "https://hotwire.io"
```

File: public/robots.txt [hunk 2]
```diff
@@ -1 +1,2 @@
 # See https://www.robotstxt.org/robotstxt.html for documentation on how to use the robots.txt file
+Sitemap: https://www.hotwire.io/sitemap.xml
```
Comment (marcoroth) [inline public/robots.txt:2]: Same here with the www, and I think this also needs to be `.xml.gz` to match the filename
```suggestion
Sitemap: https://hotwire.io/sitemap.xml.gz
```

**Review by marcoroth — COMMENTED** [review]
This is great. Thank you @joancodes! I just two small things, otherwise this looks good to me

**Review by marcoroth — APPROVED** [review]
Looks good to me, thank you for working on this @joancodes! 🚀

File: config/sitemap.rb
Comment (gismoog) [inline config/sitemap.rb:1]: Hello,

Google ignores <priority> and <changefreq> values. Google uses the <lastmod> value if it's consistently and verifiably (for example by comparing to the last modification of the page) accurate.

Comment (marcoroth) [reply config/sitemap.rb:1]: Thanks @gismoog! Do you have an idea how to properly implement this? It seems like the `lastmod` defaults to `Time.now` (at the time when the sitemap was generated).

It least this is how it looks in the current `sitemap.xml.gz`:

CleanShot 2023-10-29 at 20 02 26

**Pull request merged**